src/OrchardCore.Modules/OrchardCore.Tenants/Views/Admin/Index.cshtml

```diff
@@ -16,6 +19,14 @@
     {
         var shellSettings = shellSettingsEntry.ShellSettings;
         var host = shellSettings.RequestUrlHost?.Split(new[] { ',', ' ' }, StringSplitOptions.RemoveEmptyEntries).FirstOrDefault();
+        
+        // if first host set is not specifying a port.
+        // When developping we append the current port assuming it should never use port 80 or 443
+        if (env.EnvironmentName.Equals("development", StringComparison.OrdinalIgnoreCase) && !host.Contains(":") && Context.Request.Host.Port != 80 && Context.Request.Host.Port != 443)
```

Comment:
We can use function, and introduce static property or constants for ports. Simply we can do

```
 public int[] ReservedPorts = new [] { 80, 443 };
```

Reply:
These are methods in a `Razor View`; they can't access static properties.

Reply:
I'm starting to think that the only real solution is to use multiple hostnames with ports in the tenant configurations and to document it.